File: archiso/airootfs/usr/share/churros/preferences/pages/about.py
from widgets.page import Page
from widgets.group import Group
from widgets.row import Row

from services.about import AboutService


class AboutPage(Page):

    def __init__(self, navigator):

        super().__init__(
            navigator,
            "Acerca de",
            "Información de ChurrOS"
        )

        #
        # ChurrOS
        #

        system = Group(
            "ChurrOS"
        )

        system.add(

            Row(

                title="Distribución",

                subtitle="Sistema operativo",

                icon="system.svg",

                value=AboutService.distro()

            )

        )

        system.add(

            Row(

                title="Versión",

                subtitle="Versión instalada",

                icon="system.svg",

                value=AboutService.version()

            )

        )

        system.add(

            Row(

                title="Edición",

                subtitle="Canal de desarrollo",

                icon="system.svg",

                value=AboutService.edition()

            )

        )

        self.add(
            system
        )

        #
        # Software
        #

        software = Group(
            "Software"
        )

        software.add(

            Row(

                title="Kernel",

                subtitle="Versión del kernel",

                icon="applications.svg",

                value=AboutService.kernel()

            )

        )

        software.add(

            Row(

                title="Base",

                subtitle="Distribución base",

                icon="applications.svg",

                value=AboutService.base()

            )

        )

        software.add(

            Row(

                title="Sesión",

                subtitle="Entorno actual",

                icon="applications.svg",

                value=AboutService.session()

            )

        )

        self.add(
            software
        )

        #
        # Proyecto
        #

        project = Group(
            "Proyecto"
        )

        project.add(

            Row(

                title="Desarrollador",

                subtitle="Proyecto iniciado por",

                icon="about.svg",

                value=AboutService.developer()

            )

        )

        project.add(

            Row(

                title="Licencia",

                subtitle="Licencia del proyecto",

                icon="about.svg",

                value=AboutService.license()

            )

        )

        self.add(
            project
        )

File: rust/preferences/src/pages/about.rs
// ==========================================
// AboutPage — Acerca de ChurrOS
// (equivalente a pages/about.py)
// ==========================================

use crate::services::about::AboutService;
use crate::widgets::group::Group;
use crate::widgets::page::Page;
use crate::widgets::row::Row;

pub fn build(_navigator: gtk::Stack) -> Page {
    let page = Page::new(None, "Acerca de", Some("Información de ChurrOS"), None);

    // ChurrOS
    let mut system = Group::new("ChurrOS");

    system.add(&Row::new(
        "Distribución",
        Some("Sistema operativo"),
        Some("system.svg"),
        Some(AboutService::distro()),
        None,
        None,
    ));
    system.add(&Row::new(
        "Versión",
        Some("Versión instalada"),
        Some("system.svg"),
        Some(AboutService::version()),
        None,
        None,
    ));
    system.add(&Row::new(
        "Edición",
        Some("Canal de desarrollo"),
        Some("system.svg"),
        Some(AboutService::edition()),
        None,
        None,
    ));

    page.add(system.widget());

    // Software
    let mut software = Group::new("Software");

    software.add(&Row::new(
        "Kernel",
        Some("Versión del kernel"),
        Some("applications.svg"),
        Some(&AboutService::kernel()),
        None,
        None,
    ));
    software.add(&Row::new(
        "Base",
        Some("Distribución base"),
        Some("applications.svg"),
        Some(AboutService::base()),
        None,
        None,
    ));
    software.add(&Row::new(
        "Sesión",
        Some("Entorno actual"),
        Some("applications.svg"),
        Some(&AboutService::session()),
        None,
        None,
    ));

    page.add(software.widget());

    // Proyecto
    let mut project = Group::new("Proyecto");

    project.add(&Row::new(
        "Desarrollador",
        Some("Proyecto iniciado por"),
        Some("about.svg"),
        Some(AboutService::developer()),
        None,
        None,
    ));
    project.add(&Row::new(
        "Licencia",
        Some("Licencia del proyecto"),
        Some("about.svg"),
        Some(AboutService::license()),
        None,
        None,
    ));

    page.add(project.widget());

    page
}
